scripts/cluster_network.py

```diff
@@ -426,28 +428,23 @@ def clustering_for_n_clusters(
             f"potential_mode should be one of 'simple' or 'conservative' but is '{potential_mode}'"
         )
 
-    if custom_busmap:
-        busmap = pd.read_csv(snakemake.input.custom_busmap, index_col=0, squeeze=True)
-        busmap.index = busmap.index.astype(str)
-        logger.info(f"Imported custom busmap from {snakemake.input.custom_busmap}")
-    else:
-
-        if snakemake.config["cluster_options"]["alternative_clustering"]:
+    if not isinstance(custom_busmap, pd.Series):
+        if alternative_clustering:
             n, busmap = busmap_for_gadm_clusters(
-                n, snakemake.config["build_shape_options"]["gadm_layer_id"]
+                n, gadm_layer_id
             )  # TODO make func only return busmap, and get level from config
         else:
             busmap = busmap_for_n_clusters(
                 n, n_clusters, solver_name, focus_weights, algorithm
             )
-
-    weighted_agg_gens = True
+    else:
+        busmap = custom_busmap
 
     clustering = get_clustering_from_busmap(
         n,
         busmap,
         bus_strategies=dict(country=_make_consense("Bus", "country")),
-        aggregate_generators_weighted=weighted_agg_gens,
+        aggregate_generators_weighted=True,
```

Comment:
Have you locally tested al the three options above (alternative clustering True/False and custom_busmap as pdSeries?
I think alternative clustering False/True must be tested.
However, a note shall be added here

Reply:
alternative clustering True and False works.
Tested in a previous PR https://github.com/pypsa-meets-africa/pypsa-africa/pull/384.
What note should be added?

I didn't test the custom_busmap since we have never applied it so far.